src/main/java/org/broadinstitute/hellbender/engine/VariantWalkerGroupedByLocus.java

```diff
@@ -0,0 +1,203 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.Locatable;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.transformers.VariantTransformer;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.iterators.IntervalLocusIterator;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * VariantWalkerGroupedByLocus processes variants from a single source, either one at a time in order, or optionally
+ * grouped by locus overlap, with optional contextual information from a reference, sets of reads, and/or supplementary sources
+ * of Features. By-locus traversal is opt in, via {@link #setGroupByLocus()}, and only loci with overlapping
+ * variants are traversed.
+ *
+ * VariantWalkerGroupedByLocus authors must implement the {@link #apply} method to process each variant, and may optionally implement
+ * {@link #onTraversalStart}, {@link #onTraversalSuccess} and/or {@link #closeTool}.
+ */
+public abstract class VariantWalkerGroupedByLocus extends VariantWalkerBase {
```

Comment:
Maybe `VariantLocusWalker` would be more accurate?

Reply:
Yeah, though `VariantWalkerGroupedByLocus` has symmetry with `MultiVariantWalkerGroupedOnStart`, for what thats worth. Happy to change it if you think `VariantLocusWalker` is better .

Reply:
I think I would change it.  It's more clear that way.